### PR TITLE
Full post with comments

### PR DESCRIPTION
**Pull request by krawieck**

### CODE REVIEW

File: lib/util/hot_rank.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// import 'dart:core' show ab;
+import 'dart:math' show log, max, pow;
+
+import 'package:lemmy_api_client/lemmy_api_client.dart';
+
+double _calculateHotRank(int score, DateTime time) {
+  final elapsed = (time.difference(DateTime.now()).inMilliseconds).abs() / 36e5;
+
+  // log instead of log10 in web version. does log == log10?
```
Comment (shilangyu) [inline lib/util/hot_rank.dart:9]: Dart's `log` is base `e`, to get base 10 use: `log10(x) = log(x) / ln10`

File: lib/util/hot_rank.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// import 'dart:core' show ab;
```
Comment (shilangyu) [inline lib/util/hot_rank.dart:1]: remove unused code

File: lib/util/hot_rank.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import 'package:lemmy_api_client/lemmy_api_client.dart';
+
+double _calculateHotRank(int score, DateTime time) {
+  final elapsed = (time.difference(DateTime.now()).inMilliseconds).abs() / 36e5;
+
+  // log instead of log10 in web version. does log == log10?
+  return (10000 * log(max(1, 3 + score))) / pow(elapsed + 2, 1.8);
+}
+
+extension CommentHotRank on CommentView {
```
Comment (shilangyu) [inline lib/util/hot_rank.dart:13]: please add a comment explaining why we are calculating it ourselves

File: lib/widgets/post.dart [hunk 4]
```diff
@@ -28,26 +30,31 @@ MediaType whatType(String url) {
...
   void _goToUser() {
     print('GO TO USER');
   }
 
   void _goToPost(BuildContext context) {
-    print('GO TO POST');
+    final api = LemmyApi(instanceUrl).v1;
+    final p = api.getPost(id: post.id);
+    Navigator.of(context).push(MaterialPageRoute(
+        builder: (context) => FullPostPage.fromFuture(p, post: post)));
```
Comment (shilangyu) [inline lib/widgets/post.dart:57]: This is not Post's job to fetch full post, FullPost should do it

File: lib/pages/full_post.dart [hunk 5]
```diff
@@ -0,0 +1,99 @@
+import 'package:esys_flutter_share/esys_flutter_share.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:lemmy_api_client/src/models/post.dart';
+
+import '../widgets/comment_section.dart';
+import '../widgets/post.dart';
+
+class FullPostPage extends HookWidget {
```
Comment (shilangyu) [inline lib/pages/full_post.dart:9]: Should all pages have a `Page` suffix? If so, it has to be added to other pages

**Review by shilangyu — CHANGES_REQUESTED** [review]

File: lib/pages/full_post.dart (hunk 5 above)
Comment (krawieck) [reply lib/pages/full_post.dart:9]: yes, `Page` or `Tab` suffix

**Review by shilangyu — APPROVED** [review]

File: lib/pages/full_post.dart [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  void sharePost() => Share.text('Share post', post.apId, 'text/plain');
+
+  void savePost() {
+    //
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // final post
+    var fullPostFuture = useFuture(this.fullPost);
```
Comment (shilangyu) [inline lib/pages/full_post.dart:32]: this is an AsyncSnapshot, not a future

File: lib/pages/full_post.dart [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+    var fullPost = fullPostFuture.data;
+
+    final savedIcon = () {
+      if (fullPost != null) {
```
Comment (shilangyu) [inline lib/pages/full_post.dart:37]: use `fullPostSnap.hasData`

File: lib/pages/full_post.dart [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+      appBar: AppBar(
+        leading: BackButton(),
+        actions: [
+          IconButton(icon: Icon(Icons.share), onPressed: sharePost),
+          IconButton(icon: Icon(savedIcon), onPressed: savePost),
+          IconButton(
+              icon: Icon(Icons.more_vert), onPressed: () {}), // TODO: more menu
+        ],
+      ),
+      body: fullPost != null || post != null
```
Comment (shilangyu) [inline lib/pages/full_post.dart:66]: use `fullPostSnap.hasData`

File: lib/pages/full_post.dart [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+          // FUTURE SUCCESS
+          ? ListView(
+              physics: const AlwaysScrollableScrollPhysics(),
+              children: [
+                if (fullPost != null)
```
Comment (shilangyu) [inline lib/pages/full_post.dart:71]: use `fullPostSnap.hasData`

File: lib/pages/full_post.dart [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+                  Post(fullPost.post, fullPost: true)
+                else if (post != null)
+                  Post(post, fullPost: true)
+                else
+                  CircularProgressIndicator(),
+                if (fullPost != null)
```
Comment (shilangyu) [inline lib/pages/full_post.dart:77]: use `fullPostSnap.hasData`

File: lib/pages/full_post.dart [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+                  CommentSection(fullPost.comments,
+                      postCreatorId: fullPost.post.creatorId)
+                else
+                  Container(
+                    child: Center(child: CircularProgressIndicator()),
+                    padding: EdgeInsets.only(top: 40),
+                  ),
+              ],
+            )
+          : fullPostFuture.error != null
```
Comment (shilangyu) [inline lib/pages/full_post.dart:87]: use `fullPostSnap.hasError`

**Review by shilangyu — CHANGES_REQUESTED** [review]

**Pull request merged**